moveos/moveos/src/vm/moveos_vm.rs

```diff
@@ -317,6 +321,25 @@ where
                 )?;
                 self.execute_init_modules(init_function_modules)
             }
+        };
+
+        self.resolve_pending_init_functions()?;
+
+        action_result
+    }
+
+    /// Resolve pending init functions request registered via the NativeModuleContext.
+    fn resolve_pending_init_functions(&mut self) -> VMResult<()> {
+        let ctx = self
+            .session
+            .get_native_extensions_mut()
+            .get_mut::<NativeModuleContext>();
+        let init_functions = ctx.init_functions.clone();
+        if !init_functions.is_empty() {
+            println!("Pending init functions: {:?}", init_functions);
+            self.execute_init_modules(init_functions)
```

Comment:
For the init function of the updated module, there will be problems if it is executed again？

Reply:
It is a problem. We can make this TODO and find a solution for the module init function.

Reply:
Publish with the `rooch move publish` command only execute the init function when the module first deploys.